src/main/java/com/aws/greengrass/shadowmanager/sync/SyncHandler.java

```diff
@@ -274,4 +279,16 @@ public void pushLocalDeleteSyncRequest(String thingName, String shadowName, byte
             overallSyncStrategy.putSyncRequest(new LocalDeleteSyncRequest(thingName, shadowName, deletePayload));
         }
     }
+
+    public void setCoreThingName(String thingName) {
```

Comment:
This doesn't seem right... We are tightly coupling the `SyncHandler` to `syncOnInteractionHandler` which is a concrete implementation. This will lead to code that is harder to test and modify

Reply:
With the new implementation it's not needed, removed this part :)